.github/workflows/main.yml

```diff
@@ -158,83 +158,73 @@ jobs:
-    - name: Run tests
-      # remove this ignore when https://github.com/pschanely/hypothesis-crosshair/issues/40 is fixed
-      run: python -m pytest -W "ignore:hypothesis.internal.observability.TESTCASE_CALLBACKS is deprecated" --numprocesses auto ${{ matrix.whichtests == 'nocover' && 'hypothesis-python/tests/nocover' || 'hypothesis-python/tests/ --ignore=hypothesis-python/tests/nocover/ --ignore=hypothesis-python/tests/quality/ --ignore=hypothesis-python/tests/ghostwriter/ --ignore=hypothesis-python/tests/patching/' }}
-
-  test-osx:
-    runs-on: macos-latest
+  cross:
     strategy:
       matrix:
+        os:
+          - windows-latest
+          - macos-latest
+        python-version:
+          - "3.11"
+          - "3.13"
+        python-architecture:
+          - null
+          - "x86"
         task:
-          - check-py310-cover
-          - check-py310-nocover
-          - check-py310-niche
-          - check-py311-cover
+          - cover
+          - nocover
+          - rest
+          - alt-nocover
+          - alt-rest
+        exclude:
+          - os: macos-latest
+            python-architecture: "x86"
+          - python-version: "3.13"
+            python-architecture: "x86"
+          - python-version: "3.11"
+            task: nocover
+          - python-version: "3.11"
+            task: rest
+          - python-version: "3.13"
+            task: alt-nocover
+          - python-version: "3.13"
+            task: alt-rest
       fail-fast: false
+    runs-on: ${{ matrix.os }}
+    env:
+      PYTHONWARNDEFAULTENCODING: ""
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
-    - name: Set up Python 3.10
+    - name: Set up Python ${{ matrix.python-version }} ${{ matrix.python-architecture }}
       uses: actions/setup-python@v4
       with:
-        python-version: "3.10"
+        python-version: ${{ matrix.python-version }}
+        architecture: ${{ matrix.python-architecture }}
     - name: Restore cache
       uses: actions/cache@v3
       with:
         path: |
           ~/.cache
           ~/Library/Caches/pip
+          ~/wheelhouse
+          ~/.local
+          vendor/bundle
           .tox/
-        key: deps-${{ runner.os }}-${{ hashFiles('requirements/*.txt') }}-${{ matrix.task }}
+        key: deps-${{ runner.os }}-${{ hashFiles('requirements/*.txt') }}-${{ matrix.python-version }}-${{ matrix.python-architecture }}-${{ matrix.task }}
+        restore-keys: |
+          deps-${{ runner.os }}-${{ hashFiles('requirements/*.txt') }}-${{ matrix.python-version }}-${{ matrix.python-architecture }}
+          deps-${{ runner.os }}-${{ hashFiles('requirements/*.txt') }}
+          deps-${{ runner.os }}
+    - name: Install tox
+      run: |
+        pip install --upgrade setuptools pip wheel
+        pip install tox
     - name: Run tests
-      run: TASK=${{ matrix.task }} ./build.sh
+      working-directory: ./hypothesis-python
+      run: |
+        tox -e ${{ matrix.task }}
```

Comment:
I actually prefer this explicit `tox` invocation over a `TASK=` envvar, but I am curious: is moving to this required by this new design?